sdk-extensions/tracing-incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/trace/LeakDetectingSpanProcessor.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.trace;
+
+import static java.lang.Thread.currentThread;
+
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.internal.shaded.WeakConcurrentMap;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.lang.ref.Reference;
+import java.util.Arrays;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiConsumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A {@link SpanProcessor} which will detect spans that are never ended. It will both detect spans
+ * that are garbage collected without ever having `end()` called on them.
+ *
+ * <p>Note: using this SpanProcessor will definitely impact the performance of your application. It
+ * is not recommended for production use, as it uses additional memory for each span to track
+ * whether the span has been ended.
```

Comment:
```suggestion
 * where a leaked span was created.
```

The reality is `WeakConcurrentMap` is pretty low overhead - the main overhead of this / context debugger is tracking the stack traces.

Reply:
yeah, but those stack traces definitely add a lot of memory overhead per span, unfortunately.